blueprint/core/blueprint_ImageView.h

```diff
@@ -30,26 +32,32 @@ namespace blueprint
         {
             View::setProperty(name, value);
 
-            if (name == juce::Identifier("source"))
+            if (name == sourceProp)
             {
-                juce::String source = value.toString();
+                const juce::String source    = value.toString();
+                const juce::URL    sourceURL = source;
 
-                if (source.startsWith("data:image/"))
+                if (sourceURL.isWellFormed())
                 {
                     auto drawableImg = std::make_unique<juce::DrawableImage>();
-                    const juce::Image img = loadImageFromDataURL(source);
-                    drawableImg->setImage(img);
+                    drawableImg->setImage(loadImageFromURL(sourceURL));
                     drawable = std::move(drawableImg);
-                    return;
                 }
-
-                // handle svg image
-                drawable = std::unique_ptr<juce::Drawable>(
-                    juce::Drawable::createFromImageData(
-                        source.toRawUTF8(),
-                        source.getNumBytesAsUTF8()
-                    )
-                );
+                else if (source.startsWith("data:image/")) // juce::URL does not currently handle Data URLs
+                {
+                    auto drawableImg = std::make_unique<juce::DrawableImage>();
+                    drawableImg->setImage(loadImageFromDataURL(source));
+                    drawable = std::move(drawableImg);
+                }
+                else // If not a URL treat source prop as inline SVG/Image data
+                {
+                    drawable = std::unique_ptr<juce::Drawable>(
```

Comment:
Eventually this case should go away and this `else` block should probably be a `runtime_error`. We'll move SVG data up into the `data:` URL category

Reply:
I'll run over your earlier comments regarding this at the weekend and see if I can tidy that up.